app/controllers/concerns/recaptcha_concern.rb

```diff
@@ -5,6 +5,10 @@ module RecaptchaConcern
     before_action :allow_csp_recaptcha_src, if: :recaptcha_enabled?
   end
 
+  def recoverable_recaptcha_error?(result)
+    result.errors.keys == [:recaptcha_token]
+  end
```

Comment:
I think we could consider being more restrictive in this logic, specifically in checking that the reason for failure was due to a score falling outside the acceptable range (i.e. disallow failures due to blank or otherwise invalid tokens). One of the challenges with this is that if the user refreshes the page and resubmits the form contents, we'll likely get a `timeout-or-duplicate` error from the reCAPTCHA service. From the user's perspective, I think they should still have the chance to retry at this point. It has me wondering if should cache the token verification result in Redis for a short time, so we don't have to call out to the external service, and so that we would be able to validate it based on the initial service response.

That being said, it's probably best to save that work for a follow-on ticket.